src/NosCore.Core/Configuration/Configurator.cs

```diff
@@ -1,34 +1,47 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
 using System.Linq;
 using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.Extensions.Configuration;
 
 namespace NosCore.Core.Configuration
 {
     public static class Configurator
     {
-        public static void Configure(IConfiguration configuration, object strongTypedConfiguration)
+        private static IConfiguration ReplaceEnvironment(IConfiguration configuration, object strongTypedConfiguration)
         {
-            var regexp = new Regex(@"\${(?<variable>[a-zA-Z_]+)\s+,?\s+(?<fallback>[^}]+)}");
-            if (configuration != null)
+            var ses = configuration.GetChildren().ToList();
+            for (var index = ses.Count; index > 0; index--)
             {
-                foreach (var s in configuration.GetChildren().Where(s => s.Value != null).ToList())
+                if (ses[index - 1].Value != null)
                 {
-                    var matches = regexp.Matches(s.Value);
+                    var regexp = new Regex(@"\${(?<variable>[a-zA-Z_]+)\s*,?\s*(?<fallback>[^}]+)}");
+                    var matches = regexp.Matches(ses[index - 1].Value);
                     foreach (var match in matches.ToList())
                     {
                         var value = Environment.GetEnvironmentVariable(match.Groups[1].Value);
-                        if (value == null && (match.Groups.Count > 2) && (match.Groups[2].Value != null))
+                        if (string.IsNullOrEmpty(value) && (match.Groups.Count > 2) && (match.Groups[2].Value != null))
                         {
                             value ??= match.Groups[2].Value;
                         }
-
-                        s.Value = regexp.Replace(s.Value, value!, 1);
+                        ses[index - 1].Value = regexp.Replace(ses[index - 1].Value, value!, 1);
                     }
                 }
+                else if (ses[index - 1].GetChildren().Any())
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add the missing 'else' clause.](https://app.codacy.com/manual/NosCoreIO/NosCore/pullRequest?prid=5501796)